Implementation read/write of VOTable with Parquet serialization (#15281)
astropy/astropy: docs/whatsnew/6.0.rst

public internet connection.

.. _whatsnew-6.0-votable-parquet-serialization:

VOTable now supports PARQUET serialization

> =========================================
> The PARQUET file format allows a more efficient handling of large data amounts. However,

Reviewer: ```suggestion
=========================================

The PARQUET file format allows a more efficient handling of large data amounts. However,
```

one problem of PARQUET is that it only provides a limited number of column metadata keywords.
A way to make it consistent with VO standards is to embed it into a VOTable file.

This serialization works similar to the VOTable FITS serialization that already existed. It
basically creates two files, on VOTable file and one PARQUET file, which are linked together.
The advantage of this method is that any column metadata can be saved along with the PARQUET file,
following VO standards.

> Reading and writing of the VOTable PARQUET serialization is fully supported by the 
> `astropy.io.votable` writer and the `parse()` reader. The method works for both absolute
> and relative file paths. This serialization can be used by setting the table format attribute to votable.parquet.

Reviewer: A bit of rearranging
```suggestion
Reading and writing of the VOTable PARQUET serialization is fully supported by 
`astropy.io.votable` and the unified ``Table``  read/write interface.  This serialization can be used by setting the ``'format'`` argument to ``'votable.parquet'``. The method works for both absolute
and relative file paths.
```

See :ref:`votable_mivot` for more details.

.. _whatsnew-6.0-votable-parquet-serialization:

VOTable now supports PARQUET serialization
==========================================

The PARQUET file format allows a more efficient handling of large data amounts. However,
one problem of PARQUET is that it only provides a limited number of column metadata keywords.
A way to make it consistent with VO standards is to embed it into a VOTable file.

This serialization works similar to the VOTable FITS serialization that already existed. It
basically creates two files, on VOTable file and one PARQUET file, which are linked together.
The advantage of this method is that any column metadata can be saved along with the PARQUET file,
following VO standards.

Reading and writing of the VOTable PARQUET serialization is fully supported by
`astropy.io.votable` and the unified ``Table`` read/write interface.

> This serialization can be used by setting the ``'format'`` argument to

Reviewer 2: ```suggestion
This serialization can be used by setting the ``format`` argument to
```

``'votable.parquet'``, while ``'votable'`` can be used for reading in such a file.
The method works for both absolute and relative parquet file paths.

Example for writing:

> .. doctest-skip::

Reviewer 2: I think if you remove the `>>>`, you do not need `doctest-skip` because doctest won't pick it up, but I guess this also works.

Reviewer: This is cleaner, it shows that it's an example and we considered testing it, but decided not to, rather than it wasn't considered. IMO it also looks better.

    >>> import numpy as np
    >>> from astropy.table import Table
    >>>
    >>> # Create some fake data
    >>> number_of_objects = 10
    >>> ids = [f"COSMOS_{ii:03g}" for ii in range(number_of_objects)]
    >>> redshift = np.random.uniform(low=0, high=3, size=number_of_objects)
    >>> mass = np.random.uniform(low=1e8, high=1e10, size=number_of_objects)
    >>> sfr = np.random.uniform(low=1, high=100, size=number_of_objects)
    >>> cosmos = Table([ids, redshift, mass, sfr], names=["id", "z", "mass", "sfr"])
    >>>
    >>> # Create Column metadata
    >>> column_metadata = {
    ...    "id": {"unit": "", "ucd": "meta.id", "utype": "none"},
    ...    "z": {"unit": "", "ucd": "src.redshift", "utype": "none"},
    ...    "mass": {"unit": "solMass", "ucd": "phys.mass", "utype": "none"},
    ...    "sfr": {"unit": "solMass / yr", "ucd": "phys.SFR", "utype": "none"},
    ... }
    >>>
    >>> # Write VOTable with Parquet serialization
    >>> filename = "votable_with_parquet.vot"
    >>> cosmos.write(filename, column_metadata=column_metadata, format="votable.parquet")

> 

Reviewer 2: ```suggestion
```
